### ISSUE TITLE
Publish Castle Game Engine to Steam

### ISSUE THREAD

**Issue by shaunroselt** [castle-engine/castle-engine#480]
There are already many game engines available on Steam. It would be very cool to have Castle Game Engine on Steam too.

https://store.steampowered.com/search/?term=Game+Engine

https://store.steampowered.com/app/1670460/GameMaker/

I would love to simply be able to just launch Castle Game Engine from Steam and also see my hours spent on it through Steam.

**Comment by michaliskambi** [comment]
You're reading my mind! Or, our roadmap: https://castle-engine.io/roadmap#steam :)

This is already in progress.

- We have a Steam integration in a PR, waiting now for my review and testing: https://github.com/castle-engine/castle-engine/pull/471 . Very nice integration technologically thanks for @eugeneloza , that features latest Steamworks SDK and doesn't require any additional "wrapper" libraries. Pascal code talks to Steamworks SDK libraries directly.
- We have a page on Steam in-progress of being set up :)

Aside from easy way to install and upgrade CGE, it will give us some extra "exposure" to new users. And of course fun stuff -- like hours spend, as you write. Also achievements (e.g. "compile your first CGE game", or "see 100 compilation errors" :) ).

Here's a screenshot. This is not published yet. I want to publish it as "Early Access" before or simultaneously with CGE 7.0 release.

![cge_steam_screenshot](https://github.com/castle-engine/castle-engine/assets/1076978/3c63c6d1-1bc6-4f5f-8eef-092c3b65b836)

**Comment by shaunroselt** [comment]
That's FANTASTIC !!!!!!!!!!!!!!

I can't wait.

Achievements would be very cool too!

**Comment by shaunroselt** [comment]
It's probably going to be a lot of work, but maybe you can convince the guys over at Lazarus to publish the whole Lazarus IDE onto Steam too 😁😁😁.... or plan for it in the future.

Having both Castle Game Engine and Lazarus available on Steam would really be great!

**Comment by michaliskambi** [comment]
> You guys should make the Steam Store Page live so long and put it as "Coming Soon".

Good point. As soon as we'll have all the graphic assets nice on the store page, we'll do it :)

**Comment by shaunroselt** [comment]
@michaliskambi Any update on this?

**Comment by michaliskambi** [comment]
Not yet, sorry. Too many tasks too little time :) This is still on my high priority, alas many (too many?) things got lately on my "high priority" list. 

( Good news: these are all CGE tasks, so I'm busy working on CGE full time :) I hope to have more time in the upcoming 2 months, now that I'm done with conferences for 2023 ( https://castle-engine.io/conferences ) and articles about CGE ( https://github.com/castle-engine/bad-chess/ ) are published. )

**Comment by michaliskambi** [comment]
Our Steam integration is now part of CGE master branch, with a number of features (achievements, query various info), tested cross-platform (Windows, Linux, macOS), documented on https://castle-engine.io/steam .

Next step is to release engine on Steam -- I hope to finalize it ~next week! Stay tuned :)

**Comment by shaunroselt** [comment]
Hey @michaliskambi 

Do you by any chance have an update on the Steam Release?

**Comment by michaliskambi** [comment]
> Do you by any chance have an update on the Steam Release?

Sorry, not yet -- I started doing this, then got buried by other tasks. Thanks for the bump, I should get back to this. I'll try around July/August.

**Comment by shaunroselt** [comment]
> > Do you by any chance have an update on the Steam Release?
> 
> Sorry, not yet -- I started doing this, then got buried by other tasks. Thanks for the bump, I should get back to this. I'll try around July/August.

Hey.

Just another bump. Did you ever get to this? 😁

**Comment by shaunroselt** [comment]
> > Just another bump. Did you ever get to this? 😁
> 
> Still not, sorry -- I'm trying to do too many things... :) This is still on the roadmap, I'll get to it (and you're welcome to keep pinging me!), others are also asking about it.

Would you mind sharing what is still needed to get the game engine onto Steam in case someone wants to help? 🙏🏻
